ovsdb/client.go

```diff
@@ -90,10 +140,24 @@ func (c *Client) rpc(method string, out interface{}, args ...interface{}) error
 	req := jsonrpc.Request{
 		Method: method,
 		Params: args,
-		// Let the client handle the request ID.
+		ID:     c.requestID(),
+	}
+
+	// Add callback for this RPC ID to return results via channel.
+	ch := make(chan rpcResponse, 0)
+	c.addCallback(req.ID, ch)
+
+	if err := c.c.Send(req); err != nil {
+		return err
+	}
+
+	// Wait for callback to fire.
+	res := <-ch
```

Comment:
Do you wanna `context.Context` based timeout support here?  It's always great to have `context.Context` not only for the timeout, but also for the tracing.

Reply:
Yeah, probably a good idea, would just need to figure out how to make it work.  Can do in a follow up.

Reply:
Difinitely.  Always better to be incremental.